trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.testing;
+
+import com.google.inject.Inject;
+import com.google.inject.Scopes;
+import io.trino.aws.proxy.server.testing.TestingUtil.ForTesting;
+import io.trino.aws.proxy.server.testing.containers.S3Container.ForS3Container;
+import io.trino.aws.proxy.server.testing.harness.BuilderFilter;
+import io.trino.aws.proxy.spi.credentials.Credential;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+
+public final class RequestRewriteUtil
+{
+    public static final String TEST_CREDENTIAL_REDIRECT_BUCKET = "redirected-bucket-for-credential";
+    public static final String TEST_CREDENTIAL_REDIRECT_KEY = "redirected-key-for-credential";
+    public static final Credential CREDENTIAL_TO_REDIRECT = new Credential("credential-to-redirect", UUID.randomUUID().toString());
+
+    private RequestRewriteUtil() {}
+
+    public static class SetupRequestRewrites
+    {
+        @Inject
+        public SetupRequestRewrites(
+                TestingCredentialsRolesProvider credentialsRolesProvider,
+                @ForTesting Credentials testingCredentials,
+                @ForS3Container List<String> configuredBuckets,
+                @ForS3Container S3Client storageClient)
+        {
+            credentialsRolesProvider.addCredentials(Credentials.build(CREDENTIAL_TO_REDIRECT, testingCredentials.requiredRemoteCredential()));
+            configuredBuckets.forEach(bucket -> storageClient.createBucket(r -> r.bucket(getTargetName(bucket))));
+            storageClient.createBucket(r -> r.bucket(TEST_CREDENTIAL_REDIRECT_BUCKET));
+        }
+    }
+
+    public static class Filter
+            implements BuilderFilter
+    {
+        @Override
+        public TestingTrinoAwsProxyServer.Builder filter(TestingTrinoAwsProxyServer.Builder builder)
+        {
+            return builder
+                    .withProperty("aws.proxy.s3.path", "/api/some/s3/path")
```

Comment:
Why do we need to change the s3 path for the rewrite tests?

Reply:
This was left behind from a previous filter that needed it, good catch - removed it